scripts/bin/auto-mount

```diff
@@ -1,17 +1,80 @@
-#! /bin/sh
-
-sleep 5
-
-$SNAP/usr/local/bin/udisksctl monitor | grep --line-buffered " Device:" | while read -r -- device; do
-
-  echo "$SNAP_NAME.auto-mount found ${device}"
-  if echo $device| grep -q "[0-9]$"; then
-    DEV="$(echo $device | sed 's/^Device: //')"
-  fi
-  if [ -n "$DEV" ]; then
-    RES=`$SNAP/usr/local/bin/udisksctl mount --no-user-interaction --block-device "$DEV"`
-    # process returned value into the mount point string
-    RES2=`echo $RES | cut -f4 -d\  | cut -f1 -d.`
-    echo $RES2 > $WATCHED
-  fi
-done
+#!/bin/sh
+
+wait_for_fifo() {
+    [ ! -p "$WATCH_FILE" ] || return 0
+
+    echo "Waiting for FIFO: ${WATCH_FILE}"
+    inotifywait --quiet --csv --event create "$(dirname "$WATCH_FILE")" | \
+        while IFS=, read -r _dir _ _name; do
+            [ "${_dir}/${_name}" != "$WATCH_FILE" ] || return 0
+        done
+}
+
+watch_devices() {
+    inotifywait --quiet --monitor --event "moved_to" --event "delete" /dev/disk/by-uuid | \
+        while read -r _parent _action _uuid; do
+            case "$_action" in
+                MOVED_TO) try_mount "$_parent" "$_uuid" ;;
+                *) : ;;
```

Comment:
I don't think we need to watch the 'delete' event if we only have the `MOVED_TO` case.